src/core/TransformControls.tsx

```diff
@@ -60,8 +60,10 @@ export const TransformControls = React.forwardRef<TransformControlsImpl, Transfo
     const group = React.useRef<THREE.Group>()
 
     React.useLayoutEffect(() => {
-      if (object) controls?.attach(object instanceof THREE.Object3D ? object : object.current)
-      else controls?.attach(group.current as THREE.Object3D)
+      if (!controls) return () => {}
```

Comment:
i think it would be better writing 

```jsx
if (controls) {
  if (object) controls.attach(object instanceof THREE.Object3D ? object : object.current)
  else controls.attach(group.current as THREE.Object3D)
  return () => controls.detach()
}
```

btw i don't understand the code, how can controls ever be undefined when it's defined as:

```jsx
const [controls] = React.useState(() => new TransformControlsImpl(explCamera, gl.domElement))
```

Reply:
@drcmda Hey! This part is actually introducing issues. I solved it by using useMemo and also added a way to override the dom element. https://github.com/pmndrs/drei/pull/703

Reply:
why `if (!controls) return () => {}` is to avoid TS type error.

see:
![image](https://user-images.githubusercontent.com/6331508/148792079-830e7b47-d293-42fb-aa76-c6e502de2d24.png)

this is because the callback type is `function(): (() => TransformControlsImpl<THREE.Camera | Camera>) | undefined
`.

But now that I think about it, this was simpler.

```
if (object) controls?.attach(object instanceof THREE.Object3D ? object : object.current)
      else controls?.attach(group.current as THREE.Object3D)
      return () => {
        controls?.detach()
      }
```

Reply:
@RenaudRohlinger 
I don't noticed it is not nullish already because of your patch, thanks! so every `?` can be removed.